test/functional/fixtures/api/es-next/hooks/test.js

```diff
@@ -129,3 +130,72 @@ describe('[API] fixture.before/fixture.after hooks', () => {
         return runTests('./testcafe-fixtures/fixture-ctx.js', null, { only: 'chrome, firefox' });
     });
 });
+
+describe('[API] fixture global before/after hooks', () => {
+    before(() => {
+        global.fixtureBefore = 0;
+        global.fixtureAfter  = 0;
+    });
+
+    after(() => {
+        delete global.fixtureBefore;
+        delete global.fixtureAfter;
+    });
+
+    it('Should run hooks for all fixture', () => {
+        const hooks = {
+            fixture: {
+                before: async () => {
+                    await delay(100);
+
+                    global.fixtureBefore++;
+                },
+                after: async () => {
+                    await delay(100);
+
+                    global.fixtureAfter++;
+                },
+            },
+        };
+
+        return runTests('./testcafe-fixtures/fixture-hooks-global.js', null, { only: 'chrome', hooks });
+    });
+
+    it('Error should occur with the before hook', () => {
+        const hooks = {
+            fixture: {
+                before: {},
+                after:  async () => {
+                    await delay(100);
+
+                    global.fixtureAfter++;
+                },
+            },
+        };
+
+        return runTests('./testcafe-fixtures/fixture-hooks-global.js', null, { only: 'chrome', hooks })
```

Comment:
Can we write the server tests for these `expected type` errors instead of functional-tests?